src/main/java/com/devonfw/demoquarkus/logic/UcFindAnimalImpl.java

```diff
@@ -0,0 +1,110 @@
+package com.devonfw.demoquarkus.logic;
+
+import com.devonfw.demoquarkus.domain.model.AnimalEntity;
+import com.devonfw.demoquarkus.domain.repo.AnimalRepository;
+import com.devonfw.demoquarkus.service.mapper.AnimalMapper;
+import com.devonfw.demoquarkus.service.model.AnimalDto;
+import com.devonfw.demoquarkus.service.model.AnimalSearchCriteriaDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+@Named
+@Transactional
+@Slf4j
+public class UcFindAnimalImpl implements UcFindAnimal {
+    @Inject
+    AnimalRepository animalRepository;
+
+    @Inject
+    AnimalMapper mapper;
+
+    @Override
+    public PageImpl<AnimalDto> findAnimals(AnimalSearchCriteriaDto dto) {
```

Comment:
As @GuentherJulian evaluated [here](https://github.com/devonfw-forge/devonfw-microservices/issues/37#issuecomment-888980547), the return type has to be `PageImpl` in order to make it work properly in native mode.

Reply:
Sorry, but I do not give up too easily on this. I assume that we need to instantiate the page via `new PageImpl` but the signatures escpecially in the use-case can still remain as `PageImpl` - maybe this is not possible for the REST service in case quarkus/graalVM try to derive such infos for jackson mapping generation from signatures but I would actually doubt this unless it has been tried and confirmed. Can you double-check with @GuentherJulian  or try it yourself. If we are absolutely sure, we have to go this way but as this is kind of odd, I only want such workaround in signatures if there is no other way.